### PR TITLE
feat: add withCredentials option

### PR DESCRIPTION
**Pull request by nLight**
Enabler for the protoss project

### CODE REVIEW

File: src/XHRConnection.js [hunk 1]
```diff
@@ -128,11 +128,17 @@ export default class XHRConnection extends AbstractConnection {
   /**
    * create, prepare, open and send the xhr request
    */
-  open() {
+  open(options = {}) {
```
Comment (jaxesn) [inline src/XHRConnection.js]: Why not handle it in the constructor like the other options?

Comment (nLight) [reply src/XHRConnection.js]: it won't really matter since it's going to be hidden behind http-service

Comment (nLight) [reply src/XHRConnection.js]: though I think that could be another option

Comment (jaxesn) [reply src/XHRConnection.js]: With it being an option on open, I'm not sure how we could get that thru the connectionmanager and what not?

Comment (nLight) [reply src/XHRConnection.js]: fixed

**Review by jaxesn — DISMISSED** [review]

File: src/XHRConnection.js [hunk 2]
```diff
@@ -35,9 +35,10 @@ export default class XHRConnection extends AbstractConnection {
         Accept: "application/json"
       },
       responseType = "json",
-      timeout = 0
+      timeout = 0,
+      withCredentials = false
     } = options;
-
+
```
Comment (juliangieseke) [inline src/XHRConnection.js]: 🎨

**Comment by nLight** [comment]
done!

**Review by jaxesn — APPROVED** [review]

**Pull request merged**